### DISCUSSION TITLE
Is there a way to put text in a rectangle and have the rectangle expand with the amount of text?

### DISCUSSION THREAD

**Discussion by jsmorabito** [uxbox/uxbox#1160] [Q&A]
I know this can be achieved with [auto layout in Figma](https://help.figma.com/hc/en-us/articles/360040451373-Create-dynamic-designs-with-auto-layout) and I saw this [closed issue](https://github.com/penpot/penpot/issues/822) about auto layout here saying that new constraint features might accomplish similar functionality. I'm having trouble figuring out how to accomplish what I want if it is possible. Thanks!

**Comment by soultipsy** [comment]
Unfortunately, the effect cannot be achieved yet. The layout constraint is to let the child element follow the parent change, while the auto layout is to let the parent can adapt to the child element change, they only have a small part of the effect is overlap.

**Comment by jcklpe** [comment]
some update on what's going on: https://community.penpot.app/t/what-are-penpot-plans-for-autolayout/112

**Comment by dick-jo** [comment]
> I know this can be achieved with [auto layout in Figma](https://help.figma.com/hc/en-us/articles/360040451373-Create-dynamic-designs-with-auto-layout) and I saw this [closed issue](https://github.com/penpot/penpot/issues/822) about auto layout here saying that new constraint features might accomplish similar functionality. I'm having trouble figuring out how to accomplish what I want if it is possible. Thanks!

What you are referring to are Box Model properties ie padding and margin, they are outlined in the W3C specifications.

These are absolutely fundamental to creating UI/UX designs so I second the need for these features. This tool simply cannot serve its primary use case without them.
